src/firefly/java/edu/caltech/ipac/table/io/FITSTableReader.java

```diff
@@ -295,6 +286,154 @@ private static void logTableReadError(String fitsFilename, int tableIdx, String
                 ", HDU#: " + tableIdx + ", reason: "+reason);
     }
 
+    //This function is based on the packagedType function from the FitsStarTable class (uk.ac.starlink.fits package)
+    private static Class getClassType(Object base, int icol, boolean[] isScaled) {
+        if (base == null) {
+            if (isScaled[icol]) return Double.class;
+            return Object.class;
+        } else {
+            Class cls = base.getClass().getComponentType();
+            if (cls != null && Array.getLength(base) == 1) {
+                if (isScaled[icol]) {
+                    return Double.class;
+                }
+                switch (cls.getName()) {
+                    case "byte":
+                        return Byte.class;
+                    case "short":
+                        return Short.class;
+                    case "int":
+                        return Integer.class;
+                    case "long":
+                        return Long.class;
+                    case "float":
+                        return Float.class;
+                    case "double":
+                        return Double.class;
+                    case "boolean":
+                        return Boolean.class;
+                    case "java.lang.String":
+                        return String.class;
+                }
+            } else if (cls != null && cls.isArray()) {
+                return ArrayFuncs.flatten(base).getClass();
+            }
+            return base.getClass();
+        }
+    }
+
+    private static DataGroup readFitsTable(BasicHDU<?> hdu,
+                                          String fitsFilename,
+                                          String[] inclCols,
+                                          String[] inclHeaders,
+                                          String strategy) throws IOException, FitsException {
```

Comment:
In this new function, `inclCols` and `inclHeaders` are no longer implemented.  `metaInfo` is removed completely.  And, test is failing.
```
edu.caltech.ipac.firefly.server.util.tables.DataGroupReadTest > fitsTable FAILED
    java.lang.ClassCastException at DataGroupReadTest.java:93
```

Reply:
Yes, inclHeaders will be used for one of my TODOs...(assigning DataGroup meta at the end of this function). and I also just noticed test failing, thanks! Will look into it.